sqld/src/rpc/wal_log.rs

```diff
@@ -74,8 +85,32 @@ impl WalLog for WalLogService {
         &self,
         req: tonic::Request<LogOffset>,
     ) -> Result<tonic::Response<Self::LogEntriesStream>, Status> {
+        let replica_addr = req
+            .remote_addr()
+            .ok_or(Status::internal("No remote RPC address"))?;
+        {
+            let guard = self.replicas_with_hello.read().await;
+            if !guard.contains(&replica_addr) {
+                return Err(Status::failed_precondition("Hello message missing"));
```

Comment:
can we use something that look more like an error code, like `"NO_HELLO"`, and put that in a `const` so we can reuse that to check for the code on the client side